toolsrc/src/vcpkg/base/system.cpp

```diff
@@ -39,7 +39,7 @@ namespace vcpkg
             case CPUArchitecture::ARM: return "arm";
             case CPUArchitecture::ARM64: return "arm64";
             case CPUArchitecture::S390X: return "s390x";
-            default: Checks::exit_with_message(VCPKG_LINE_INFO, "unexpected vcpkg::System::CPUArchitecture");
+            default: Checks::exit_maybe_upgrade(VCPKG_LINE_INFO, "unexpected vcpkg::System::CPUArchitecture");
```

Comment:
```suggestion
            default: Checks::unexpected(VCPKG_LINE_INFO);
```

This is an enumeration, if a value is not in the list then that's a coding error.